src/classes/socket.ts

```diff
@@ -4,7 +4,12 @@ import Emitter from "./emitter.ts";
 const OPENED = Symbol("Opened");
 const CLOSED = Symbol("Closed");
 
-export default class Socket extends Emitter {
+export default class Socket extends Emitter<{
+	"message": [MessageEvent<string>];
```

Comment:
100% correct types here.

Reply:
I tried this for my implementation as well. However, it seems like the types for our different environments don't overlap enough, hence I did this for now: https://github.com/surrealdb/surrealdb.js/pull/44/files#diff-bb3aaab9ff8f3973e54e438135d58c54bd2376b0424ff46f256290c3270f1807R11-R33

Reply:
What is different in ws module? I think it has the same fields so it is a MessageEvent<string>

Reply:
And it is not exposed so doesnt matter.